src/rascal/math/bessel.hh

```diff
@@ -100,7 +100,7 @@ namespace rascal {
        *         Note that a reference is returned to avoid unnecessary
        *         copies.
        */
-      Eigen::Ref<Eigen::ArrayXXd> get_values() {
+      auto get_values() {
```

Comment:
Please don't do this. I want to see what is going on just by looking at the function signature, not the implementation.

Reply:
This is not the type of the returned value I and actually don't really know what it is since `ref(this->bessel_values.leftCols(this->l_max + 1));` is returned. This is a perfect case to use auto I think

Reply:
I would rather take the time to figure the type once, so that future readers of the code don't have to do it in the future. You can get the compiler to print you the type with

```
auto a = ref(this->bessel_values.leftCols(this->l_max + 1));
decltype(a)::do_not_exists = 44;
``` 

This will generate an error message with the compiler-deduced type.

Reply:
That is the problem and your code snippet is not applicable here. 
In principle this function returns 2 different types since in the else branch it returns `Eigen::Ref<Eigen::ArrayXXd>`. So Eigen here is doing its magic finding a common ground which is equivalent to `Eigen::Ref<Eigen::ArrayXXd>`making sure it compiles properly.

Reply:
This is confusing me. The magic Eigen type must still be nameable, right? You say it is not `Eigen::Ref<Eigen::ArrayXXd>`, is it closer to `Eigen::Ref<Eigen::Map<Eigen::ArrayXXd>>`?

Reply:
Related, if the return type of the new branch is convertible to `Eigen::Ref<Eigen::ArrayXXd>`, why not keep the interface as returning Eigen::Ref<Eigen::ArrayXXd>?

Reply:
For reference, the actual type was `Eigen::Ref<const Eigen::ArrayXXd>`, fiexed in a4c433cba32d7c8e804fe0729bb9dcc767e79177